src/orion/core/worker/warm_start/multi_task_wrapper.py

```diff
@@ -0,0 +1,245 @@
+        """Number of trials suggested by the algorithm **in the target task**"""
+        # TODO: Unsure about this.
+        return super().n_suggested
+
+    @property
+    def n_observed(self):
+        """Number of completed trials observed by the algorithm"""
+        # TODO: Unsure about this.
+        return super().n_observed
+
+    def set_state(self, state_dict):
+        """Reset the state of the algorithm based on the given state_dict
+
+        :param state_dict: Dictionary representing state of an algorithm
+        """
+        super().set_state(state_dict)
+
+    @property
+    def trials_from_target_task(self) -> Iterable[Trial]:
+        return (trial for trial in self.algorithm.registry if get_task_id(trial) == 0)
+
+    @property
+    def trials_from_other_tasks(self) -> Iterable[Trial]:
+        return (trial for trial in self.algorithm.registry if get_task_id(trial) != 0)
+
+    @property
+    def is_done(self):
+        """Return True, if an algorithm holds that there can be no further improvement.
+        By default, the cardinality of the specified search space will be used to check
+        if all possible sets of parameters has been tried.
+        """
+        # NOTE: DEBUGGING:
+        total_trials = len(self.unwrapped.registry)
+        trials_from_target_task = sum(
+            get_task_id(trial) == 0 for trial in self.unwrapped.registry
+        )
+        trials_from_other_tasks = total_trials - trials_from_target_task
+        log.debug(
+            f"Trials from target task: {trials_from_target_task}, "
+            f"trials from other tasks: {trials_from_other_tasks} "
+        )
+        log.debug(f"self.n_observed: {self.n_observed}")
+        log.debug(f"self.n_observed: {self.n_suggested}")
+        log.debug(f"wrapped algo.is_done: {self.algorithm.is_done}")
+
+        # FIXME: Do the same logic as in the BaseAlgorithm.is_done, but only consider trials
+        # from the target task.
+        if trials_from_target_task >= self.space.cardinality:
+            return True
+
+        max_trials = getattr(self, "max_trials", float("inf"))
+        if trials_from_target_task >= max_trials:
+            return True
+
+        return False
+
+    @contextmanager
+    def in_task(self, task_id: int):
+        """Contextmanager that temporarily changes the value of `self.current_task_id`
+        to `task_id` and restores the original value after exiting the with block.
+        """
+        previous_task_id = self.current_task_id
+        self.current_task_id = task_id
+        yield
+        self.current_task_id = previous_task_id
+
+    def _add_task_id(self, trial: Trial, task_id: int) -> Trial:
+        # NOTE: Need to do a bit of gymnastics here because the points are expected
+        # to be tuples. In order to add the task id at the right index, we convert
+        # them like so:
+        # Trial -> dict (.params) -> dict (add task_id) -> Trial.
+        params = trial.params.copy()
```

Comment:
Maybe this could reuse/be merged in part with `_copy_status_and_results` from the base algo wrapper module?

Reply:
Not sure I understand how though. I'm already using `_copy_status_and_results` below.
I made this a bit more explicit using the `trial_with_params` and `trial_with_info` arguments, and removed the outdated comments in 3ae6acb5b196cb4a3d373828ba740bf3728b0be4.